full-node/sov-ethereum/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+use borsh::ser::BorshSerialize;
+use const_rollup_config::ROLLUP_NAMESPACE_RAW;
+use demo_stf::app::DefaultPrivateKey;
+use demo_stf::runtime::{DefaultContext, Runtime};
+use ethers::types::transaction::eip2718::TypedTransaction;
+use ethers::types::Bytes;
+use ethers::utils::rlp::Rlp;
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::core::params::ArrayParams;
+use jsonrpsee::http_client::{HeaderMap, HttpClient};
+use jsonrpsee::RpcModule;
+use jupiter::da_service::DaServiceConfig;
+use sov_evm::call::CallMessage;
+use sov_evm::evm::EvmTransaction;
+use sov_modules_api::transaction::Transaction;
+
+const GAS_PER_BYTE: usize = 120;
+
+pub fn get_ethereum_rpc(config: DaServiceConfig) -> RpcModule<Ethereum> {
+    let mut rpc = RpcModule::new(Ethereum { config });
+    register_rpc_methods(&mut rpc).expect("Failed to register sequencer RPC methods");
+    rpc
+}
+
+pub struct Ethereum {
+    config: DaServiceConfig,
+}
+
+impl Ethereum {
+    fn make_client(&self) -> HttpClient {
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            "Authorization",
+            format!("Bearer {}", self.config.celestia_rpc_auth_token.clone())
+                .parse()
+                .unwrap(),
+        );
+
+        jsonrpsee::http_client::HttpClientBuilder::default()
+            .set_headers(headers)
+            .max_request_body_size(default_max_response_size()) // 100 MB
+            .build(self.config.celestia_rpc_address.clone())
+            .expect("Client initialization is valid")
+    }
+
+    async fn send_tx_to_da(
+        &self,
+        raw: Vec<u8>,
+    ) -> Result<serde_json::Value, jsonrpsee::core::Error> {
+        let blob = vec![raw].try_to_vec().unwrap();
+        let client = self.make_client();
+        let fee: u64 = 2000;
+        let namespace = ROLLUP_NAMESPACE_RAW.to_vec();
+        let gas_limit = (blob.len() + 512) * GAS_PER_BYTE + 1060;
+
+        let mut params = ArrayParams::new();
+        params.insert(namespace)?;
+        params.insert(blob)?;
+        params.insert(fee.to_string())?;
+        params.insert(gas_limit)?;
+        client
+            .request::<serde_json::Value, _>("state.SubmitPayForBlob", params)
+            .await
+    }
+}
+
+fn register_rpc_methods(rpc: &mut RpcModule<Ethereum>) -> Result<(), jsonrpsee::core::Error> {
+    rpc.register_async_method(
+        "eth_sendRawTransaction",
+        |parameters, ethereum| async move {
+            let data: Bytes = parameters.one().unwrap();
+            let data = data.as_ref();
+
+            // todo handle panics and unwraps.
+            if data[0] > 0x7f {
+                panic!("Invalid transaction type")
```

Comment:
Woud `anyhow::bail` work here?

Reply:
Here we will need to craft a proper `jsonrpsee` response, the same way Ethereum does it. But it is out of scope for now.